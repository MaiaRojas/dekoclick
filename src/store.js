'use strict';


import { createStore, applyMiddleware, compose } from 'redux';
import { reactReduxFirebase } from 'react-redux-firebase';
import thunk from 'redux-thunk';
import api from './middleware/api';
import reducers from './reducers';


const middlewares = [thunk, api];


// Add redux Firebase to compose
const createStoreWithFirebase = compose(
  reactReduxFirebase({
    apiKey: process.env.FIREBASE_API_KEY,
    authDomain: `${process.env.FIREBASE_PROJECT}.firebaseapp.com`,
    databaseURL: `https://${process.env.FIREBASE_PROJECT}.firebaseio.com`,
    projectId: process.env.FIREBASE_PROJECT,
    storageBucket: `${process.env.FIREBASE_PROJECT}.appspot.com`,
    messagingSenderId: process.env.FIREBASE_MESSAGING_SENDER_ID,
  }, { userProfile: 'users' }),
)(createStore);


if (process.env.NODE_ENV !== 'production') {
  const { logger } = require('redux-logger');
  middlewares.push(logger);
}


export default createStoreWithFirebase(reducers, applyMiddleware(...middlewares));
