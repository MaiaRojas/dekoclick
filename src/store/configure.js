'use strict';


import { createStore, applyMiddleware } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import thunk from 'redux-thunk';
import api from '../middleware/api';
import rootReducer from '../reducers';


const configure = (preloadedState, history) => {

  // Build the middleware for intercepting and dispatching navigation actions
  const middlewares = [routerMiddleware(history), thunk, api];

  if (process.env.NODE_ENV === 'development') {
    const { logger } = require('redux-logger');
    middlewares.push(logger);
  }

  return createStore(
    rootReducer,
    preloadedState,
    applyMiddleware(...middlewares)
  );
};


export default configure;
