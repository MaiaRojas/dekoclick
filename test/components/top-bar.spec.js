import React from 'react';
import { render, shallow } from 'enzyme';
import sinon from 'sinon';
import TopBar from '../../src/components/top-bar';
import { Provider } from 'react-redux';
import store from '../../src/store';


describe('<TopBar />', () => {

  it('should print warning in console.error when missing title', () => {
    const stub = sinon.stub(console, 'error');
    const component = render(
      <Provider store={store}>
        <TopBar />
      </Provider>
    );
		expect(stub.getCalls().length > 0).toBe(true);
    expect(stub.getCall(0).args[0]).toMatchSnapshot();
    stub.restore();
	});

  it('should render to a <header> containing an <h2> with the title', () => {
    const component = render(
      <Provider store={store}>
        <TopBar title="Blah blah blah" />
      </Provider>
    );
    expect(component.get(0).tagName).toBe('header');
    expect(component.hasClass('mui-fixed')).toBe(true);
    expect(component.find('h2').text()).toBe('Blah blah blah');
  });

  it('should have classes prop injected and title prop on instance', () => {
    const component = render(
      <Provider store={store}>
        <TopBar title="foo" />
      </Provider>
    );
    expect(component).toHaveLength(1);
    const $h2 = component[0].children[0].children[1];
    expect(component[0].attribs.class).toMatch(/TopBar/);
    expect($h2.children[0].type).toBe('text');
    expect($h2.children[0].data).toBe('foo');
  });

});
