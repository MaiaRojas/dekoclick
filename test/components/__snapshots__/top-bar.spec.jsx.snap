// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> should print warning in console.error when missing title 1`] = `
Array [
  "Warning: Failed prop type: The prop \`title\` is marked as required in \`TopBar\`, but its value is \`undefined\`.
    in TopBar
    in WithStyles(TopBar)
    in Connect(WithStyles(TopBar))
    in Provider",
]
`;
